notebook/services/contents/manager.py

```diff
@@ -317,22 +317,31 @@ def increment_filename(self, filename, path='', insert=''):
             The name of a file, including extension
         path : unicode
             The API path of the target's directory
+        insert: unicode
+            The characters to insert after the base filename
 
         Returns
         -------
         name : unicode
             A filename that is unique, based on the input filename.
         """
-        path = path.strip('/')
-        basename, ext = os.path.splitext(filename)
+        # Extract the full suffix from the filename (e.g. .tar.gz)
+        dirname = os.path.dirname(filename)
+        basename = os.path.basename(filename)
```

Comment:
This can be summarised as `dirname, basename = os.path.split(filename)`.

Also, I think we should leave in the `.strip('/')` that was there before, so that passing in `foo/bar/` gives you `foo/bar1`, not `foo/bar/1`.